chain/jsonrpc/res/validator.html

```diff
@@ -0,0 +1,417 @@
+<html>
+
+<head>
+    <style>
+        table {
+            width: 100%;
+            border-collapse: collapse;
+        }
+
+        table,
+        th,
+        td {
+            border: 1px solid black;
+        }
+
+        td {
+            text-align: left;
+            vertical-align: top;
+            padding: 8px;
+        }
+
+        th {
+            text-align: center;
+            vertical-align: center;
+            padding: 8px;
+            background-color: lightgrey;
+        }
+
+        tr.active {
+            background-color: #eff8bf;
+        }
+
+        .approval-skip {
+            background-color: yellow;
+        }
+
+        .approval-target-rollback {
+            background-color: red;
+        }
+
+        .approval-ok {
+            background-color: lightgreen;
+        }
+
+        .client-actor-delayed {
+            font-size: x-large;
+            font-weight: bold;
+        }
+
+        .validator-missing {
+            background-color: grey;
+        }
+
+        .validator-endorse {
+            background-color: lightgreen;
+        }
+
+        .validator-skip-0 {
+            background-color: #75f09e33;
+        }
+
+        .validator-skip-1 {
+            background-color: #ffce5633;
+        }
+
+        .validator-skip-2 {
+            background-color: #36a2eb33;
+        }
+
+        .validator-skip-3 {
+            background-color: #4bc0c033;
+        }
+
+        .validator-skip-4 {
+            background-color: #ff641e33;
+        }
+
+        .chunk-delay-orange {
+            background-color: orange;
+        }
+
+        .chunk-delay-red {
+            background-color: red;
+        }
+    </style>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/chart.js"></script>
+
+    <script>
+        // draw the colorful bar with the approvals.
+        // approval_map should contain the mapping from 'skips' (int) to number of votes, skip == 0 means that this is an endorsement.
+        function drawApprovals(approvalMap, missingCount, context) {
```

Comment:
Heh, I feel like random bits of non-trivial JS in embedded HTML files run a risk of turning into unmaintainable mess over time. But this is probably still fine at this stage.